front/tests/utils/api-setup.ts

```diff
@@ -20,6 +16,13 @@ export const getApiRequest = async (
   return response.json();
 };
 
+// Helper function to handle API error responses
+export function handleApiResponse(response: APIResponse, errorMessage: string) {
```

Comment:
I think I would still extract the `!response.ok` outside of the function and use it like this : 

```suggestion
  if (errorMessage && !response.ok) {
```

Because we name the function `handleErrorMessage` but if there is no error, it doesn't do anything

Reply:
Add response check and a default message